webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/batch/PurgeOldInventoryFiles.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ *************************************************************************************
+ * Copyright 2022 Normation SAS
+ *************************************************************************************
+ *
+ * This file is part of Rudder.
+ *
+ * Rudder is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In accordance with the terms of section 7 (7. Additional Terms.) of
+ * the GNU General Public License version 3, the copyright holders add
+ * the following Additional permissions:
+ * Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+ * Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+ * Public License version 3, when you create a Related Module, this
+ * Related Module is not considered as a part of the work and may be
+ * distributed under the license agreement of your choice.
+ * A "Related Module" means a set of sources files including their
+ * documentation that, without modification of the Source Code, enables
+ * supplementary functions or services in addition to those offered by
+ * the Software.
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+package com.normation.rudder.batch
+
+import better.files.File
+import com.normation.errors.IOResult
+import com.normation.inventory.domain.InventoryProcessingLogger
+import com.normation.utils.CronParser._
+import com.normation.zio._
+import cron4s.CronExpr
+import java.time.Instant
+import zio._
+import zio.clock.Clock
+import zio.duration._
+
+/**
+ * A scheduler which delete old inventory files under /var/rudder/inventories/{received, failed}
+ * It use a cron-like config parsed by cron4s.
+ */
+class PurgeOldInventoryFiles(
+    optCron:          Option[CronExpr],
+    maxAge:           Duration,
+    cleanDirectories: List[File]
+) {
+
+  val logger = InventoryProcessingLogger
+
+  // check is a file is older than max age (true) or not. Only true for regular file.
```

Comment:
right, the name is not the best, but in the context of purging inventory files, it's not so bad - but I agree, it needs conventional knowledge, so rise the threshold for contrib etc. 
Plus, we need to filter out directories in any cases, because Java nio API does not provide a consistent creation time for directories and leads to surprising results